src/main/kotlin/uk/gov/dwp/dataworks/controllers/UserContainerController.kt

```diff
@@ -23,14 +26,16 @@ class UserContainerController {
     ])
     @PostMapping("/deployusercontainers")
     fun launchTask(@RequestBody requestBody: Model){
-        taskDeploymentService.taskDefinitionWithOverride(
-                requestBody.ecsClusterName,
-                requestBody.emrClusterHostName,
-                requestBody.albName ,
-                requestBody.userName,
-                requestBody.containerPort,
-                requestBody.jupyterCpu,
-                requestBody.jupyterMemory
-        )
+        if (!existingUserServiceCheck.check(requestBody.userName, requestBody.ecsClusterName)){
```

Comment:
It would look nicer here to reverse the `if` statement and probably log a message to say that nothing has been actioned.

Reply:
This way we negate the need for an additional else statement and, as far a I can see, the log would be confirming a user environment exists, which is the endpoint either way. We could look at returning something to the front end service in this case - as the user will not need to wait.

Reply:
Not necessarily, the following could be done:
```
        if (existingUserServiceCheck.check(requestBody.userName, requestBody.ecsClusterName)){
            logger.log("Redirecting user to running containers, as they exist")
            return
        }
        taskDeploymentService.taskDefinitionWithOverride(
                requestBody.ecsClusterName,
                requestBody.emrClusterHostName,
                requestBody.albName ,
                requestBody.userName,
                requestBody.containerPort,
                requestBody.jupyterCpu,
                requestBody.jupyterMemory
        )
```
I also think the endpoint is not designed to only confirm if an endpoint exists and it is useful information to log out that you are not creating more resources but instead directing the user to existing ones. This might help anyone debugging in future as it informs them what logical actions the system is taking